lib/std/build.zig

```diff
@@ -1939,6 +1947,40 @@ pub const LibExeObjStep = struct {
         self.step.dependOn(&artifact.step);
     }
 
+    /// The value is the path in the cache dir.
+    /// Adds a dependency automatically.
+    /// basename refers to the basename of the WriteFileStep
+    pub fn addBuildOptionWriteFile(
+        self: *LibExeObjStep,
+        name: []const u8,
+        write_file: *WriteFileStep,
+        basename: []const u8,
+    ) void {
+        self.build_options_write_file_args.append(.{
+            .name = name,
+            .write_file = write_file,
+            .basename = basename,
+        }) catch unreachable;
+        self.step.dependOn(&write_file.step);
+    }
+
+    /// Adds build option corresponding to the absolute (resolved) path.
+    /// Path is either an absolute path or a path relative to the build
+    /// working directory.
+    pub fn addPathBuildOption(
+        self: *LibExeObjStep,
+        name: []const u8,
+        path: []const u8,
+    ) void {
+        // Note that pathFromRoot uses resolve path, so this will have
+        // correct behavior even if getOutputPath is already absolute.
+        const escaped_abs_path = self.builder.fmt(
+            "{s}",
+            std.zig.fmtEscapes(self.builder.pathFromRoot(path)),
+        );
```

Comment:
This will cause the path to be double escaped since `addBuildOption` already handles string escaping.

This whole function seems a bit unnecessary since all it does is `.addBuildOption([]const u8, name, b.pathFromRoot(path))` which I think is more explicit about what value you are getting.

Reply:
Oh yeah, that's why I didn't orginally use fmtEscapes, forgot about that. I agree that the function isn't needed. When I was originally writing this would need to handle a bit more and then I never reconsidered.

Reply:
I removed it in the last commit.